### DISCUSSION TITLE
Is there any way to use the t function of 'next-i18next' in the ts file?

### DISCUSSION THREAD

**Discussion by zjy365** [i18next/next-i18next#2275] [Q&A]
Is there any way to use the t function of 'next-i18next' in the ts file?
I want to support internationalization by defining some json fragments in a non-hooks environment, that is, in a ts file.

**Comment by adrai** [comment]
That's conceptually not possible without knowing the user's language... usually coming via http request... so without an async handling function part you could only use some default translations...
